addons/survey/tests/test_survey_invite.py

```diff
@@ -172,22 +165,29 @@ def test_survey_invite_public(self):
         self.assertEqual(answers.mapped('partner_id'), self.customer)
 
     @users('survey_manager')
-    def test_survey_invite_token(self):
+    def test_survey_invite_token_internal(self):
         Answer = self.env['survey.user_input']
 
-        self.survey.write({'access_mode': 'token'})
+        self.survey.write({'access_mode': 'token', 'users_login_required': True})
         action = self.survey.action_send_survey()
         invite_form = Form(self.env[action['res_model']].with_context(action['context']))
 
-        invite_form.partner_ids.add(self.customer)
-        invite_form.emails = 'test1@example.com, Raoulette Vignolette <test2@example.com>'
+        with self.assertRaises(UserError):  # do not allow to add customer (partner without user)
+            invite_form.partner_ids.add(self.customer)
+        with self.assertRaises(UserError):  # do not allow to add portal user
+            invite_form.partner_ids.add(self.user_portal.partner_id)
+        invite_form.partner_ids.clear()
+        invite_form.partner_ids.add(self.user_emp.partner_id)
+        with self.assertRaises(UserError):
+            invite_form.emails = 'test1@example.com, Raoulette Vignolette <test2@example.com>'
+        invite_form.emails = False
 
         invite = invite_form.save()
         invite.action_invite()
 
         answers = Answer.search([('survey_id', '=', self.survey.id)])
-        self.assertEqual(len(answers), 3)
+        self.assertEqual(len(answers), 1)
         self.assertEqual(
             set(answers.mapped('email')),
-            set(['test1@example.com', 'Raoulette Vignolette <test2@example.com>', self.customer.email]))
-        self.assertEqual(answers.mapped('partner_id'), self.customer)
+            set([self.user_emp.email]))
```

Comment:
@tde-banana-odoo typo or something I'm missing here?

Reply:
ActuallyI swapped two methods (to have an alaphabetical order as this is the test order) and I think the diff is really not clear.